mne/label.py

```diff
@@ -897,6 +906,138 @@ def write_label(filename, label, verbose=None):
+    elif label.subject is None:
+        pass
+    elif subject != label.subject:
+        raise ValueError("The label specifies a different subject (%r) from "
+                         "the subject parameter (%r)."
+                         % label.subject, subject)
+
+    return label, subject, subjects_dir
+
+
+def _assign_split_membership(label_divs, neighbors):
+    """Helper to assign membership of a list of nodes to label components"""
+
+    # Find overlap between new node neighbors and existing label divisions
+    membership = []
+    for di, div in enumerate(label_divs):
+        if len(neighbors.intersection(div)) > 0:
+            membership.append(di)
+
+    # If node connects to an existing div, assign it to that div
+    if len(membership) > 0:
+        label_divs[membership[0]].update(neighbors)
+
+        # If node connects to 2+ divs, combine them all into one division
+        # (Could also do this once after all nodes memberships are compiled)
+        if membership > 1:
+            for mi in sorted(membership[1:], reverse=True):
+                label_divs[membership[0]].update(label_divs.pop(mi))
+
+    # If edge doesn't connect to existing div, create a new one
+    else:
+        label_divs.append(neighbors)
+
+
+def _split_label_contig(label_to_split, subject=None, subjects_dir=None):
+    """Split label into contiguous regions (i.e., connected components)
+
+    Parameters
+    ----------
+    label_to_split : Label | str
+        Label which is to be split (Label object or path to a label file).
+    subject : None | str
+        Subject which this label belongs to (needed to locate surface file;
+        should only be specified if it is not specified in the label).
+    subjects_dir : None | str
+        Path to SUBJECTS_DIR if it is not set in the environment.
+
+    Returns
+    -------
+    labels : list of Label
+        The contiguous labels, in order of decending size.
+    """
+    # Convert to correct input if necessary
+    label_to_split, subject, subjects_dir = _prep_label_split(label_to_split,
+                                                              subject,
+                                                              subjects_dir)
+
+    # Find the spherical surface to get vertices and tris
+    surf_fname = '.'.join((label_to_split.hemi, 'sphere'))
+    surf_path = op.join(subjects_dir, subject, 'surf', surf_fname)
+    surface_points, surface_tris = read_surface(surf_path)
+
+    label_divs = []  # List of contiguous sets of dipoles/nodes
+    verts = set(label_to_split.vertices)
+    # Compute edges from `tris` and take upper triangle to avoid repeat edges
+    edges = sparse.triu(mesh_edges(surface_tris), format='csr')
+
+    # Throw out edges if both verts don't belong to the label
+    coo_edges = edges.tocoo()
+    for (e_x, e_y) in zip(coo_edges.row, coo_edges.col):
+        if len(verts.intersection(set([e_x, e_y]))) < 2:
+            edges[e_x, e_y] = 0
+    edges.eliminate_zeros()
```

Comment:
why don't you use scipy

http://docs.scipy.org/doc/scipy-0.17.0/reference/generated/scipy.sparse.csgraph.connected_components.html

?